File: src/MainService.Models/Entities/UserMedicalInsuranceCompany.cs
namespace MainService.Models.Entities
{
    public class UserMedicalInsuranceCompany
    {
        public int UserId { get; set; }
        public AppUser User { get; set; } = null!;
        public int MedicalInsuranceCompanyId { get; set; }
        public MedicalInsuranceCompany MedicalInsuranceCompany { get; set; } = null!;

        public int? DocumentId { get; set; }
        public Document? Document { get; set; }

        public bool IsMain { get; set; } = false;
        public string? PolicyNumber { get; set; }
    }
}

File: client/src/app/_models/users/userMedicalInsuranceCompany/userMedicalInsuranceCompany.ts
import { SelectOption } from "src/app/_models/base/selectOption";
import { Document } from "src/app/_models/documents/document";


export class UserMedicalInsuranceCompany {
  id: number | null = null;
  isMain: boolean | null = false;
  policyNumber: string | null = null;
  medicalInsuranceCompany: SelectOption | null = null;
  document: Document = new Document();

  constructor(init?: Partial<UserMedicalInsuranceCompany>) {
    Object.assign(this, init);
  }
}
